src/components/Files/PreUploadModal.tsx

```diff
@@ -0,0 +1,413 @@
+    useState(false);
+  const [selectedFolderId, setSelectedFolderId] = useState<string | undefined>(
+    undefined,
+  );
+
+  const { refs, context } = useFloating({
+    open: isOpen,
+    onOpenChange: onClose,
+  });
+  const role = useRole(context);
+  const dismiss = useDismiss(context, { outsidePressEvent: 'mousedown' });
+  const { getFloatingProps } = useInteractions([role, dismiss]);
+  const headingId = useId();
+  const descriptionId = useId();
+
+  const folderPath = useMemo(() => {
+    return (
+      getParentAndCurrentFoldersById(folders, selectedFolderId)
+        .map((folder) => folder.name)
+        .reverse()
+        .join('/') || undefined
+    );
+  }, [folders, selectedFolderId]);
+
+  const handleSelectFiles = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      setErrorMessage('');
+
+      const files = Array.from(
+        (e.target as HTMLInputElement).files as FileList,
+      );
+
+      let incorrectSizeFiles: string[] = [];
+      let incorrectTypeFiles: string[] = [];
+      const filteredFiles = files.reduce<File[]>((acc, file) => {
+        const fileType = file.name.slice(file.name.lastIndexOf('.'));
+        if (file.size > 512 * bytesInMb) {
+          incorrectSizeFiles = incorrectSizeFiles.concat(file.name);
+          return acc;
+        }
+        if (
+          !allowedTypes.includes('*/*') &&
+          allowedTypes.length &&
+          !allowedTypes.includes(fileType)
+        ) {
+          incorrectTypeFiles = incorrectTypeFiles.concat(file.name);
+          return acc;
+        }
+
+        return acc.concat(file);
+      }, []);
+      if (incorrectSizeFiles.length > 0) {
+        setErrorMessage(
+          (oldMessage) =>
+            oldMessage +
+            '\n' +
+            t(
+              `Max file size up to 512 Mb. Next files haven't been uploaded: `,
+            ) +
+            incorrectSizeFiles.join(', '),
+        );
+      }
+      if (incorrectTypeFiles.length > 0) {
+        setErrorMessage(
+          (oldMessage) =>
+            oldMessage +
+            '\n' +
+            t(
+              `Supported types: {{allowedTypes}}. Next files haven't been uploaded: `,
+              { allowedTypes: allowedTypes.join(', ') },
+            ) +
+            incorrectTypeFiles.join(', '),
+        );
+      }
+
+      setSelectedFiles((oldFiles) =>
+        oldFiles.concat(
+          filteredFiles.map((file) => ({
+            fileContent: file,
+            id: file.name,
+            name: file.name,
+          })),
+        ),
+      );
+      if (uploadInputRef.current) {
+        uploadInputRef.current.value = '';
+      }
+    },
+    [allowedTypes, t],
+  );
+
+  const handleUpload = useCallback(() => {
+    if (attachments.length + selectedFiles.length > 10) {
+      setErrorMessage(
+        t(
+          `Maximum allowed attachments number is {{maxAttachmentsAmount}}. With your uploadings amount will be {{selectedAttachmentsAmount}}`,
```

Comment:
How are we going to use such long text as key for others locales (i18n)?

Reply:
We don't have any other locales for now, so when we will work on that we will need to refactor all such texts and replace them with keys, but for now we are not doing that for simplicity